app/models/manageiq/providers/base_manager/metrics_capture.rb

```diff
@@ -1,8 +1,171 @@
+    targets.each_slice(query_size).each do |target_group|
+      start_range, end_range, counters_data = capture_targets(target_group, interval_name, start_time, end_time)
+      perf_process_queue(ems, "perf_process", interval_name, start_range, end_range, counters_data)
+    end
+  end
+
+  # Perform capture on list of targets
+  # @param targets [Array<Object>] array of targets to be captured
+  # @param start_time [String, nil] Start time
+  # @param end_time   [String, nil] Ending time
+  # [{
+  #   :ems_id         => ems_id,         # "2"
+  #   :ems_ref        => target.ems_ref, # "vm-55"
+  #   :ems_klass      => ("Vm", "Host", "Storage")
+  #   :interval_name  => ("realtime", "hourly")
+  #   :start_range    => start_time,
+  #   :end_range      => end_time,
+  #   :counters       => {
+  #     "#{counter_id}_#{instance}" => { # e.g.: "143_" =>
+  #         :counter_key           => :"#{group}_#{name}_#{stats}_#{rollup}", # :net_usage_rate_average
+  #         :rollup                => ("realtime", ...)
+  #         :precision             => (0.1, 1),
+  #         :unit_key              => ("percent", "kiloBytesPerSecond"),
+  #         :vim_key               => counter_id, # "143"
+  #         :instance              => instance,   # ""
+  #         :capture_interval      => interval,   # "20"
+  #         :capture_interval_name => ("realtime", "hourly"),
+  #       }
+  #   },
+  #   :counter_values => {
+  #     timestamp => {                           # "2017-09-25T19:55:20Z"
+  #       "#{counter_id}_#{instance}" => Numeric # "143_" => 0
+  #     }
+  #   }
+  # }]
+  # @return [Array<Hash{Symbol=>Integer,String,Array<String>,Hash{String=>Float}}>
+
+  def capture_targets(targets, interval_name, start_time, end_time)
+    start_time, end_time = fix_capture_start_end_time(interval_name, start_time, end_time)
+
+    all_start_range  = nil
+    all_end_range    = nil
+    all_counter_data = {}
+    targets.each do |target|
+      start_range, end_range, counter_data = target.just_perf_capture(interval_name, start_time, end_time)
+      all_counter_data.merge!(counter_data)
+      all_start_range = start_range if all_start_range.nil? || all_start_range > start_range
+      all_end_range = end_range if all_end_range.nil? || all_end_range < end_range
+    end
+
+    [all_start_range, all_end_range, all_counter_data]
+  end
+
+  def fix_capture_start_end_time(interval_name, start_time = nil, end_time = nil)
+    start_time = start_time.utc unless start_time.nil?
+    end_time = end_time.utc unless end_time.nil?
+
+    # Determine the start_time for capturing if not provided
+    if interval_name == 'historical'
+      start_time ||= Metric::Capture.historical_start_time
+    else
+      start_time ||= ems.last_metrics_success_date
+      # For hourly on the first capture, we don't want to get all of the
+      #   historical data, so we shorten the query
+      start_time ||= 4.hours.ago.utc if interval_name == 'hourly'
```

Comment:
We should probably move `4.hours.ago.utc` as the "realtime cutoff" to a constant since we use it [here](https://github.com/ManageIQ/manageiq/blob/master/app/models/metric/ci_mixin/capture.rb#L59) too

Reply:
fixed